### PR TITLE
no-jira: add release workflow

### PR DESCRIPTION
**Pull request by ColinDKelley**
Github workflow copied from `invoca-env`.

### CODE REVIEW

File: .github/workflows/gem_release.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        run: |
+          echo ::set-output name=current_version::${GITHUB_REF#refs/tags/v}
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Get Changelog Entry
+        id: changelog_reader
+        uses: mindsers/changelog-reader-action@v1
+        with:
+          version: ${{ steps.tag_name.outputs.current_version }}
+        path: ./CHANGELOG.md
```
Comment (jebentier) [inline .github/workflows/gem_release.yml:26]: This needs to be indented one more I think

Comment (jebentier) [reply .github/workflows/gem_release.yml:26]: Yea I believe so, this is an argument for the step which goes under the `with` section. [Like this](https://github.com/Invoca/invoca-service_discovery/blob/main/.github/workflows/gem_release.yml#L27)

Comment (ColinDKelley) [reply .github/workflows/gem_release.yml:26]: Yeah, `invoca-service_discovery` definitely has it indented that way. So change both?

Comment (ColinDKelley) [reply .github/workflows/gem_release.yml]: Ah, I hadn't pulled `invoca-env` recently. That's already been fixed there. And now it's fixed here.  🎉

**Review by jebentier — APPROVED** [review]

**Pull request merged**